src/convert.jl

```diff
@@ -264,14 +280,16 @@ function convert(::Type{@jimport(java.util.List)}, x::Vector, V::Type{JavaObject
 end
 
 # Convert a reference to a java.lang.String into a Julia string. Copies the underlying byte buffer
-function unsafe_string(jstr::JString)  #jstr must be a jstring obtained via a JNI call
-    if isnull(jstr); return ""; end #Return empty string to keep type stability. But this is questionable
+unsafe_string(jstr::JString) = unsafe_string(jstr.ptr)  #jstr must be a jstring obtained via a JNI call
+
+function unsafe_string(jstr::Ptr{Nothing})  #jstr must be a jstring obtained via a JNI call
+    if jstr == C_NULL; return ""; end #Return empty string to keep type stability. But this is questionable
     pIsCopy = Array{jboolean}(undef, 1)
     buf::Ptr{UInt8} = ccall(jnifunc.GetStringUTFChars, Ptr{UInt8},
-                            (Ptr{JNIEnv}, Ptr{Nothing}, Ptr{jboolean}), penv, jstr.ptr, pIsCopy)
+                            (Ptr{JNIEnv}, Ptr{Nothing}, Ptr{jboolean}), penv, jstr, pIsCopy)
     s = unsafe_string(buf)
     ccall(jnifunc.ReleaseStringUTFChars, Nothing, (Ptr{JNIEnv}, Ptr{Nothing}, Ptr{UInt8}), penv,
-          jstr.ptr, buf)
+          jstr, buf)
     return s
 end
```

Comment:
@aviks Could we incorporate the changes to this file convert.jl into JavaCall proper?

Reply:
Sure. Is the current code likely a bug?

Reply:
I think it was just quoted out of context. What I meant is that most of the changes to convert.jl seem acceptable to merge. It will have to be updated for the new JNI module though.